zookeeper-server/src/test/java/org/apache/zookeeper/server/NettyServerCnxnTest.java

```diff
@@ -110,6 +113,22 @@ public void testSendCloseSession() throws Exception {
         }
     }
 
+    /**
+     * In the {@link #setUp()} routine, the maximum number of connections per IP
+     * is set to 1. This tests that if more than one connection is attempted, the
+     * connection fails.
+     */
+    @Test(timeout = 40000, expected = ProtocolException.class)
+    public void testMaxConnectionPerIpSurpased() throws Exception {
+        Assert.assertTrue(
```

Comment:
```java
assertThat(subClass, instanceOf(BaseClass.class));
```

Reply:
Shouldn't we test this in the setUp() method instead of inside a single test?

Reply:
It is not common to add tests into a setup() method.  Better to put here and if it fails, developer can easily pin-point the issue and add a breakpoint for debug.